iroha_client/tests/subscribe_to_changes.rs

```diff
@@ -1,43 +1,97 @@
+            object: AssetDefinition::new(asset_definition_id),
+            destination_id: domain_name.to_string(),
+        };
+        let mut iroha_client = Client::new(&ClientConfiguration::from_iroha_configuration(
+            &configuration,
+        ));
+        task::block_on(iroha_client.submit(create_asset.into())).expect("Failed to prepare state.");
+        while let Some(change) = stream.next().await {
             println!("Change received {:?}", change);
-        } else {
-            panic!("Failed to receive change.");
+            match change {
+                Occurrence::Created(entity)
+                | Occurrence::Updated(entity)
+                | Occurrence::Deleted(entity) => match entity {
+                    Entity::Block(_) => {
+                        println!("Entity changed: {:?}", entity);
+                        return ();
+                    }
+                    _ => println!("Received not expected change: {:?}", entity),
+                },
+            }
         }
+        panic!("Failed to receive change.");
     }
 
-    fn create_and_start_iroha() {
-        let temp_dir = TempDir::new().expect("Failed to create TempDir.");
+    #[async_std::test]
+    async fn client_subscribe_to_transaction_changes_request_should_receive_transaction_change() {
+        thread::spawn(|| {
+            let temp_dir = TempDir::new().expect("Failed to create TempDir.");
+            let mut configuration = Configuration::from_path(CONFIGURATION_PATH)
+                .expect("Failed to load configuration.");
+            configuration.torii_configuration.torii_url = "127.0.0.1:1338".to_string();
+            configuration.torii_configuration.torii_connect_url = "127.0.0.1:8890".to_string();
+            configuration
+                .kura_configuration
+                .kura_block_store_path(temp_dir.path());
+            let iroha = Iroha::new(configuration.clone());
+            task::block_on(iroha.start()).expect("Failed to start Iroha.");
+            //Prevents temp_dir from clean up untill the end of the tests.
+            #[allow(clippy::empty_loop)]
+            loop {}
+        });
+        thread::sleep(Duration::from_millis(300));
         let mut configuration =
             Configuration::from_path(CONFIGURATION_PATH).expect("Failed to load configuration.");
-        configuration
-            .kura_configuration
-            .kura_block_store_path(temp_dir.path());
-        let iroha = Iroha::new(configuration.clone());
-        task::block_on(iroha.start()).expect("Failed to start Iroha.");
+        configuration.torii_configuration.torii_url = "127.0.0.1:1338".to_string();
+        configuration.torii_configuration.torii_connect_url = "127.0.0.1:8890".to_string();
+        let mut iroha_client = Client::with_maintenance(
+            &ClientConfiguration::from_iroha_configuration(&configuration),
+        );
+        let mut stream = iroha_client
+            .subscribe_to_block_changes()
```

Comment:
The client subscribes to block changes, but the test is about transaction changes. And later entity is parsed to transaction. This seems a bit confusing. Maybe naming can be improved.